app/jobs/upload_users_job.rb

```diff
@@ -0,0 +1,24 @@
+class UploadUsersJob < ApplicationJob
+  USER_FIELDS = [:user_name, :last_name, :first_name, :id_number, :email].freeze
+
+  def self.on_complete_js(_status)
+    'window.location.reload.bind(window.location)'
+  end
+
+  def perform(user_class, data, encoding)
+    progress.total = data.lines.count
+    user_class.transaction do
+      MarkusCsv.parse(data, encoding: encoding, skip_blanks: true, row_sep: :auto) do |row|
+        user_data = {}
+        USER_FIELDS.each_with_index do |field, index|
```

Comment:
You don't need a separate variable for these fields. Instead, rely on `EndUser::CSV_ORDER` to determine the order of the values in the row.

You can use the `zip` method to iterate over `EndUser::CSV_ORDER` and `row` in parallel.